docs/manual.html

```diff
@@ -144,13 +144,16 @@ <h4>Finishing parsing</h4>
 <h4>Constructor</h4>
 
 <dl class="reference">
-	<dt><strong>lxp.new(<em>callbacks [, separator[, merge_character_data]]</em>)</strong></dt>
+	<dt><strong>lxp.new(<em>callbacks [, separator[, merge_character_data, [attribute_position]]]</em>)</strong></dt>
```

Comment:
should we change this function signature to an options table, instead of positional args?

Reply:
That would probably be a more user friendly API, but it would also be a major breaking change. This way the change is non-breaking and people should be able to upgrade without a hitch. The option to change the signature at a later date when we're considering cutting a major release would still be open too. Or provide a new function name for the new signature. Whatever the long term API design would be I'd be in favor of a non-breaking change we can get through and widely distributed over a major version bump that breaks things with little motivation for people to take the dive.

```suggestion
	<dt><strong>lxp.new(<em>callbacks [, separator[, merge_character_data[, attribute_position]]]</em>)</strong></dt>
```

Reply:
No need to be breaking. Change the signature to `new(callbacks[, options])`, and just check if `options` is a string (old format, then also grab `merge_character_data`), or if it is a table, in which case the options are in the table.
This is probably easier done Lua side than C side.

Reply:
Re-opening the comment (it got closed when accepting @alerque 's suggested change).

Changing the signature is doable (I'm unsure what you mean with "This is probably easier done Lua side than C side", both are doable and there's no Lua side for this call currently, so the C side is likely easier...). On the other hand, shouldn't it be an issue of its own?

Reply:
good point, let's close this